assets/js/app/Panel.js

```diff
@@ -190,16 +230,33 @@ LocusZoom.Panel.prototype.initialize = function(){
 };
 
 
-// Create a new data layer by data layer class
-LocusZoom.Panel.prototype.addDataLayer = function(DataLayerClass){
-    if (typeof DataLayerClass !== "function"){
-        return false;
+// Create a new data layer by layout object
+LocusZoom.Panel.prototype.addDataLayer = function(id, layout){
+    if (typeof id !== "string"){
+        throw "Invalid data layer id passed to LocusZoom.Panel.prototype.addDataLayer()";
+    }
+    if (typeof layout !== "object"){
+        throw "Invalid data layer layout passed to LocusZoom.Panel.prototype.addDataLayer()";
+    }
+    if (typeof this.data_layers[layout.id] !== "undefined"){
+        throw "Cannot create data layer with id [" + id + "]; data layer with that id already exists";
     }
-    var data_layer = new DataLayerClass();
+    if (typeof layout.class !== "string" || typeof LocusZoom[layout.class] !== "function"){
```

Comment:
I'm a bit worried about looking on the main LocusZoom object for resolving layout classes (`LocusZoom[layout.class]`) Especially if a user wanted to add their own they could over right a function we need for something else? Perhaps we could use another custom collection class here with add/get/set methods.